### ISSUE TITLE
app.getLocale() on MacOS returns wrong value in latest electron versions

### ISSUE THREAD

**Issue by MMhunter** [electron/electron#26612]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  *   9.3.4, 10.1.16
* **Operating System:**
  * Mac OS 10.13, 10.14, 10.15, 11.0
* **Last Known Working Electron version:** 
  * 9.3.3, 10.1.15

### Expected Behavior

app.getLocale() should return correct locale.`zh-CN`, e.g.

### Actual Behavior

app.getLocale() always returns 'en-US'.

### To Reproduce

````javascript
const app = require('electron')

app.whenReady().then(()=>{
  console.log(app.getLocale());
});

````

**Comment by ckerr** [comment]
Possibly related to https://github.com/electron/electron/pull/26185 whose backports both landed in 9.3.4 (#26242) and 10.1.6 (#26241).

Another locale PR https://github.com/electron/electron/pull/26185 is also in 10.1.6, but wasn't merged into 9-x-y until a week after v9.3.4 was released, and so can probably be ruled out

**Comment by jamiebullock** [comment]
I can reproduce this in Electron 9.3.4 on macOS 11.0.1

**Comment by JoaquinScript** [comment]
Can repro on electron 10.1.3

**Comment by popod** [comment]
@ckerr any chance to have a fix for this issue with electron 9 and 10 ? Thanks

For all with this issue, it seems that `getLocale()` works again fine since electron v11.1.1 !

**Comment by 1111mp** [comment]
Can repro on electron 9.4.0 on macOS 11.2.1

**Comment by Wonder233** [comment]
Can repro on electron 10.4.0 on macOS 10.15.7

**Comment by nornagon** [comment]
Looks like this was fixed in the range https://github.com/electron/electron/compare/v10.4.2...v11.0.0-nightly.20200525, which is somewhat perplexing, as none of those commits seem to touch anything locale-related.

**Comment by csett86** [comment]
@nornagon One explanation why you won't find a fix to it between 10.4.2 and 11.0.0: This was fine in both 10.0.0 and 11.0.0, but current assumtion is that it broke with a backport in 10 and 9 as mentioned by @ckerr:

Possibly related to #26185 whose backports both landed in 9.3.4 (#26242) and 10.1.6 (#26241).

**Comment by nornagon** [comment]
@csett86 the above fiddle fails on 10.4.2 and passes on 11.0.0. So I don't believe those backports are related.

**Comment by mahnunchik** [comment]
I've faced with wrong locale on 10.4.2.

**Comment by nornagon** [comment]
@mahnunchik yep, exactly. It's broken in 10.4.2 and fixed in 11.0.0.

**Comment by csett86** [comment]
Or to put it another way: It was working with 10.0.0, broken somewhere during 10 maintenance, probably at 10.1.6. It is working with 11 and later.

**Comment by nornagon** [comment]
Confirmed it broke in https://github.com/electron/electron/compare/v10.1.5...v10.1.6, so https://github.com/electron/electron/pull/26241 seems like a likely culprit. It's still unclear to me why it's fixed in v11.0.0-nightly.20200525 though as the range https://github.com/electron/electron/compare/v10.4.2...v11.0.0-nightly.20200525 contains nothing obviously related.

**Comment by csett86** [comment]
From my understanding 11.0.0 and the betas of 11 where locale was always working were released before 10.1.6. (11-beta1 was released in August, 11.0.0 on 16 Nov 2020 and 10.1.6 on 18 Nov 2020).

So from my point of view there is no commit on 11 that fixes it, as it was never broken and thus never fixed. Its only broken on the 10 and 9 maintenance branches.

But as of today I would suggest the sensible way is to upgrade Electron to 11+, as that has to happen for all apps anyway sooner than later.

**Comment by nornagon** [comment]
@csett86 ah I see, so that range `v10.4.2...v11.0.0-nightly.20200525` is a bit misleading then, as it doesn't take into account the fact that v10 and v11 diverged before that point. So perhaps v10 missed a backport that needed to go along with #26241.

In any case, as @csett86 mentioned, v10 is currently at "security fixes only" & is nearing end-of-life. I don't think this is going to be fixed in v10 unfortunately. I'm going to close this as it's fixed in v11+.
